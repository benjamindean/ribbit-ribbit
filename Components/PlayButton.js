// @flow
import React, { PureComponent } from 'react';
import { TouchableNativeFeedback, View } from 'react-native';
import PropTypes from 'prop-types';
import { Entypo } from '@expo/vector-icons';
import colors from '../Styles/colors';
import styles from '../Styles/main';

export default class PlayButton extends PureComponent {
	render() {
		return (
			<TouchableNativeFeedback
				background={TouchableNativeFeedback.Ripple(colors.ripple, true)}
				onPressIn={this.props.onPressIn}
				onPressOut={this.props.onPressOut}>
				<View style={styles.recordButton}>
					<Entypo name={this.props.name} size={this.props.size} color={colors.black} />
				</View>
			</TouchableNativeFeedback>
		);
	}
}

PlayButton.defaultProps = {
	size: 60,
	name: 'controller-record'
};

PlayButton.propTypes = {
	onPressIn: PropTypes.func.isRequired,
	onPressOut: PropTypes.func.isRequired,
	name: PropTypes.string,
	size: PropTypes.number
};
