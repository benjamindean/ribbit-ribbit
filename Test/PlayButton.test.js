import React from 'react';
import PlayButton from '../Components/PlayButton';

import renderer from 'react-test-renderer';

it('renders without crashing', () => {
	const rendered = renderer
		.create(<PlayButton onPressIn={() => {}} onPressOut={() => {}} name={'controller-play'} />)
    .toJSON();
    
	expect(rendered).toBeTruthy();
});
