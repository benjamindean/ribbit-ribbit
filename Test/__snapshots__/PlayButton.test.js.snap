// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#96A892",
        "borderColor": "rgba(0, 0, 0, 0.1)",
        "borderRadius": 125,
        "borderWidth": 20,
        "height": 250,
        "justifyContent": "center",
        "width": 250,
      },
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
</View>
`;
